### DISCUSSION TITLE
Speed up UBL mesh probing

### DISCUSSION THREAD

**Discussion by gothcha** [Jyers/Marlin#1236] [General Discussion]
I have found that enabling UBL_HILBERT_CURVE in Configuration.h improves the point to point moves that cause the probe to move erratically around the bed during the mesh probing.

#define UBL_HILBERT_CURVE       // Use Hilbert distribution for less travel when probing multiple points

With this enabled it takes a more systematic path and takes far less time.

Maybe this should be enabled by default?

**Comment by Jyers** [comment]
Intesteing, yeah I'll take a look at it

**Reply by gothcha** [reply]
Jacob another idea, I have modified my firmware to display the mesh automatically after mesh creation and after auto tilt current mesh. I found it a nuisance to go have to go through two menus to see the resultant mesh.
Plus I moved the mesh viewer to be direct from the main levelling menu and moved the display values and asymmetric setting to the leveling settings menu. Thus eliminating one menu layer.

**Reply by Jyers** [reply]
Hmmm, perhaps that would be better...
